code/modules/food_and_drinks/food/snacks_pie.dm

```diff
@@ -43,8 +47,19 @@
 		reagents.reaction(hit_atom, TOUCH)
 	if(isliving(hit_atom))
 		var/mob/living/L = hit_atom
-		if(stunning)
-			L.Paralyze(20) //splat!
+		if(L.client && H)
+			var/experience_given = 30
+			if(L.GetComponent(/datum/component/creamed))
+				experience_given *= 0.1
+			if(HAS_TRAIT(L.mind, TRAIT_LAW_ENFORCEMENT_METABOLISM))
+				experience_given *= 3
+			H.mind.adjust_experience(/datum/skill/pie_throwing, experience_given)
+		if(stunning && H)
+			var/skillmod = H.mind.get_skill_speed_modifier(/datum/skill/pie_throwing)
+			if(skillmod > 25)// journeyman level or higher
+				L.Paralyze(skillmod) //splat!
+			else
+				L.Stun(skillmod) //splish!
```

Comment:
What you are thinking of is Knockdown and Paralyze respectively. Paralyze is actually a harder stun than Stun.

(Stun is the thing goliaths do btw)